### PR TITLE
Remove OpenCensus tracer from Istio XDS

### PR DESCRIPTION
**Pull request by lei-tang**
**Please provide a description of this PR:**

OpenCensus is deprecated. This PR removes the OpenCensus tracer from Istio XDS.

Related issue: https://github.com/istio/istio/issues/50808.

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Dual Stack
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [x ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/networking/core/tracing.go [hunk 1]
```diff
@@ -200,14 +195,6 @@ func configureFromProviderConfig(pushCtx *model.PushContext, proxy *model.Proxy,
 			}
 			return otelLightStepConfig(clusterName, hostname, provider.Lightstep.GetAccessToken())
 		}
-	case *meshconfig.MeshConfig_ExtensionProvider_Opencensus:
```
Comment (howardjohn) [inline pilot/pkg/networking/core/tracing.go:203]: we should probably add some form of warnings if these are used. 

Maybe in manifests/charts/istio-control/istio-discovery/templates/NOTES.txt for helm and operator/pkg/apis/istio/v1alpha1/validation/validation.go for operator?

can be a followup

**Review by howardjohn — COMMENTED** [review]
LG

File: tools/packaging/common/envoy_bootstrap.json [hunk 2]
```diff
@@ -736,57 +736,6 @@
...
-  {{- else if .stackdriver }}
-  ,
-  "tracing": {
-    "http": {
-      "name": "envoy.tracers.opencensus",
-      "typed_config": {
-      "@type": "type.googleapis.com/envoy.config.trace.v3.OpenCensusConfig",
-      "stackdriver_exporter_enabled": true,
-      "stackdriver_project_id": "{{ .stackdriverProjectID }}",
-      {{ if .sts_port }}
```
Comment (howardjohn) [inline tools/packaging/common/envoy_bootstrap.json:748]: Do we only use sts_port for stackdriver? If so, can clean up that code in istio-agent as well in the future

cc @kyessenov 

(note: the STS I cleaned up was separate, we had 2 STS things in the agent)

Comment (kyessenov) [reply tools/packaging/common/envoy_bootstrap.json:748]: It's different impersonations I believe. Yeah, we can move it to Google agent as well.

Comment (lei-tang) [reply tools/packaging/common/envoy_bootstrap.json:748]: STS is also used for Google CA. STS clean up can be a separate work.

File: pilot/pkg/networking/core/tracing.go (hunk 1 above)
Comment (lei-tang) [reply pilot/pkg/networking/core/tracing.go:203]: Ack.

File: tools/packaging/common/envoy_bootstrap.json (hunk 2 above)
Comment (howardjohn) [reply tools/packaging/common/envoy_bootstrap.json:748]: Google CA's usage of STS is in https://github.com/istio/istio/pull/51043

File: pilot/pkg/networking/core/tracing.go (hunk 1 above)
Comment (zirain) [reply pilot/pkg/networking/core/tracing.go:203]: a warn log is adoptable.

**Review by zirain — APPROVED** [review]
LGTM

**Comment by lei-tang** [comment]
@howardjohn Can you review/approve this PR? Thanks!

**Review by howardjohn — APPROVED** [review]

**Pull request merged**